devicetypes/smartthings/zll-rgb-bulb.src/zll-rgb-bulb.groovy

```diff
@@ -67,27 +83,46 @@ def parse(String description) {
 	if (finalResult) {
 		log.debug finalResult
 		sendEvent(finalResult)
-	}
-	else {
+	} else {
 		def zigbeeMap = zigbee.parseDescriptionAsMap(description)
 		log.trace "zigbeeMap : $zigbeeMap"
 
 		if (zigbeeMap?.clusterInt == COLOR_CONTROL_CLUSTER) {
-			if(zigbeeMap.attrInt == ATTRIBUTE_HUE){  //Hue Attribute
+			if(zigbeeMap.attrInt == ATTRIBUTE_HUE && shouldUseHueSaturation()){ //Hue Attribute
 				def hueValue = Math.round(zigbee.convertHexToInt(zigbeeMap.value) / 0xfe * 100)
 				sendEvent(name: "hue", value: hueValue, displayed:false)
-			}
-			else if(zigbeeMap.attrInt == ATTRIBUTE_SATURATION){ //Saturation Attribute
+			} else if(zigbeeMap.attrInt == ATTRIBUTE_SATURATION && shouldUseHueSaturation()) { //Saturation Attribute
 				def saturationValue = Math.round(zigbee.convertHexToInt(zigbeeMap.value) / 0xfe * 100)
 				sendEvent(name: "saturation", value: saturationValue, displayed:false)
+			} else if(zigbeeMap.attrInt == ATTRIBUTE_X) { //X Attribute
+				state.currentRawX = zigbee.convertHexToInt(zigbeeMap.value)
+				//log.debug "xValue = $state.currentRawX"
+			} else if(zigbeeMap.attrInt == ATTRIBUTE_Y) { //Y Attribute
+				state.currentRawY = zigbee.convertHexToInt(zigbeeMap.value)
+				//log.debug "yValue = $state.currentRawY"
 			}
-		}
-		else {
+
```

Comment:
Nit: extra line

Reply:
I think diff view made this look weird. :/